python/xnvme-cy-bindings/setup.py

```diff
@@ -0,0 +1,178 @@
+import glob
+import os
+import re
+import shutil
+import subprocess
+import sysconfig
+from pathlib import Path
+
+from setuptools import Extension, find_namespace_packages, setup
+
+try:
+    from aux.generate_cython_bindings import generate_pyx
+    from Cython.Build import cythonize
+
+    cython = True
+except ImportError:
+    cython = False
+
+xnvme_init_py_path = "xnvme/__init__.py"
+try:
+    os.remove(
+        xnvme_init_py_path
+    )  # Don't misinterpret the local directory as the xnvme package
+except FileNotFoundError:
+    pass
+import xnvme.cython_header  # noqa: E402
+
+PACKAGE_NAME = "xnvme"
+cy_xnvme_pkg_path = os.path.dirname(os.path.realpath(__file__))
+WORKSPACE_ROOT = cy_xnvme_pkg_path + "/../.."
```

Comment:
Given above, this becomes `cy_xnvme_pkg_path.parent.parent`  -- avoids os-specific path-mangling.